### DISCUSSION TITLE
Requesting Guidance on Configuring Lock Duration and Max Stalled Count in Bull Configuration

### DISCUSSION THREAD

**Discussion by Jayshri-Intlon** [taskforcesh/bullmq#2492] [Q&A]
**Issue**
In our project, we are looking to optimize our task management system to ensure smoother workflow processes. As part of this optimization effort, we need to configure two key parameters: lockDuration and maxStalledCount.

**Code snippet**
```
import { RedisConfigService } from '../app/config/redis-config.service';
import type { RegisterQueueOptions } from '@nestjs/bullmq';
import { BullModule } from '@nestjs/bullmq';

const redisConfigService = new RedisConfigService(process.env);
const bullOptions: RegisterQueueOptions = {
  name: 'entityCreateQueue',
  connection: redisConfigService.bullConfig.connection,
};

@Module({
  imports: [
    TypeOrmModule.forFeature([Entity, EntityStatusHistory]),
    BullModule.registerQueue(bullOptions),
  ],
```

**Comment by manast** [comment]
I see, but what is the question?

**Reply by Jayshri-Intlon** [reply]
In my project, I have a job queue named 'entityCreateQueue' that runs periodically to complete certain tasks within a specific timeframe. However, I've noticed that sometimes it generates duplicates shortly after completing a task. To address this issue, I've encountered recommendations to set a 'lockDuration' with a larger value and a 'maxStalledCount' to 0, as mentioned in the references I've consulted. However, I'm uncertain about where and how to configure these settings within my Bull queue in the Nest JS application.

- [Bull Issue #1591](https://github.com/OptimalBits/bull/issues/1591)
- [BullMQ WorkerOptions](https://api.docs.bullmq.io/interfaces/v5.WorkerOptions.html#maxStalledCount)
